### PR TITLE
Fixes for multiple abap-on-prem new system prompts

### PR DESCRIPTION
**Pull request by IainSAP**
https://github.com/SAP/open-ux-tools/issues/2193

Fixes for :

- failed basic auth allows user to continue
- changing sap-client does not re-authenticate
- new system name is allowing an empty string which breaks storage
- new system names gets overwritten when changing client or system url

### CODE REVIEW

File: packages/odata-service-inquirer/src/prompts/connectionValidator.ts [hunk 1]
```diff
@@ -387,11 +441,18 @@ export class ConnectionValidator {
                 isSystem,
                 odataVersion
             });
-
+            LoggerHelper.logger.debug(`ConnectionValidator.validateUrl() - status: ${status}; url: ${url}`);
+            // Since an exception was not thrown, this is a valid url
+            this.validity.urlFormat = true;
+            this._validatedUrl = url;
```
Comment (longieirl) [inline packages/odata-service-inquirer/src/prompts/connectionValidator.ts:447]: Do we need to ensure that even if a http 200 is returned, it's not a saml HTML response.

Comment (IainSAP) [reply packages/odata-service-inquirer/src/prompts/connectionValidator.ts:447]: I think that should be dealt with by a lower level module, i.e. axios extension?

**Review by longieirl — APPROVED** [review]
Changes look good.
Test coverage is good.
Changeset is present and correct.

**Review by korotkovao — APPROVED** [review]
Looks good to me, thank you @IainSAP

**Pull request merged**